File: src/main/java/xyz/d1snin/corby/model/Cooldown.java
/*                          GNU GENERAL PUBLIC LICENSE
 *                            Version 3, 29 June 2007
 *
 *        Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
 *            Everyone is permitted to copy and distribute verbatim copies
 *             of this license document, but changing it is not allowed.
 */

package xyz.d1snin.corby.model;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import net.dv8tion.jda.api.entities.User;
import xyz.d1snin.corby.commands.Command;

@EqualsAndHashCode
@Getter
@Setter
@AllArgsConstructor
public class Cooldown {
  private User user;
  private int cooldown;
  private Command command;
}

File: src/main/kotlin/xyz/d1snin/corby/model/Cooldown.kt
/*
 * BSD 3-Clause License, Copyright (c) 2021, Corby and contributors, All rights reserved.
 */

package xyz.d1snin.corby.model

import net.dv8tion.jda.api.entities.User
import xyz.d1snin.corby.commands.AbstractCommand

data class Cooldown(
    val user: User,
    val command: AbstractCommand
) {
    var cooldown = command.cooldown
}
